lib/lighthouse/benefits_documents/form526/update_documents_status_service.rb

```diff
@@ -64,30 +64,52 @@ def update_documents_status
 
       def update_document_status(status)
         document_upload = @lighthouse526_document_uploads.find_by!(lighthouse_document_request_id: status['requestId'])
+        statsd_document_base_key(STATSD_DOCUMENT_TYPE_KEY_MAP[document_upload.document_type])
 
-        # UploadStatusUpdater encapsulates all parsing of a status response from Lighthouse
-        status_updater = BenefitsDocuments::Form526::UploadStatusUpdater.new(status, document_upload)
-        status_updater.update_status
-
-        statsd_document_type_key = STATSD_DOCUMENT_TYPE_KEY_MAP[document_upload.document_type]
-        statsd_document_base_key = "#{STATSD_BASE_KEY}.#{statsd_document_type_key}"
+        status_updater(status, document_upload)
+        @status_updater.update_status
 
         if document_upload.completed?
           # ex. 'api.form526.lighthouse_document_upload_processing_status.bdd_instructions.complete'
-          StatsD.increment("#{statsd_document_base_key}.#{STATSD_DOCUMENT_COMPLETE_KEY}")
+          StatsD.increment("#{@statsd_document_base_key}.#{STATSD_DOCUMENT_COMPLETE_KEY}")
         elsif document_upload.failed?
-          # Because Lighthouse's processing steps are subject to change, these metrics must be dyanmic.
-          # Currently, this should return either claims_evidence or benefits_gateway_service
-          failure_step_key = status_updater.get_failure_step.downcase
-
-          # ex. 'api.form526.lighthouse_document_upload_processing_status.bdd_instructions.failed.claims_evidence'
-          StatsD.increment("#{statsd_document_base_key}.#{STATSD_DOCUMENT_FAILED_KEY}.#{failure_step_key}")
-        elsif status_updater.processing_timeout?
+          log_failure(document_upload)
+        elsif @status_updater.processing_timeout?
           # Triggered when a document is still pending more than 24 hours after processing began
           # ex. 'api.form526.lighthouse_document_upload_processing_status.bdd_instructions.processing_timeout'
-          StatsD.increment("#{statsd_document_base_key}.#{STATSD_PROCESSING_TIMEOUT_KEY}")
+          StatsD.increment("#{@statsd_document_base_key}.#{STATSD_PROCESSING_TIMEOUT_KEY}")
         end
       end
+
+      def status_updater(status, document_upload)
+        # UploadStatusUpdater encapsulates all parsing of a status response from Lighthouse
+        @status_updater ||= BenefitsDocuments::Form526::UploadStatusUpdater.new(status, document_upload)
+      end
+
+      def statsd_document_base_key(statsd_document_type_key)
+        @statsd_document_base_key ||= "#{STATSD_BASE_KEY}.#{statsd_document_type_key}"
+      end
+
+      def log_failure(document_upload)
```

Comment:
Nice. I like that you pulled out some logic from`update_document_status` and made these methods 🎉